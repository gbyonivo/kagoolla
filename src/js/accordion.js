import {
  getElementById,
  getElementsByClass,
  createElement,
  addElementsToAccordion,
  addElementsToElement
} from '../functions/document';

export const onClickAccordion = (id) => {
  const content = getElementById(`accordion${id}`);
  if (!content) return;
  if (content.classList.contains('active')) {
    content.classList.remove('active');
    return;
  }
  const accordions = getElementsByClass('accordion');
  accordions.forEach((accordion) => {
    accordion.classList.remove('active');
  });
  content.classList.add('active');
};

export const beerDetailsBodyContent = (beer) => {
  const tagLine = createElement('div', { class: 'tagLine' }, `<em>tag line:</em><span>${beer.tagline}</span>`);
  const description = createElement('div', { class: 'description' }, `<em>description:</em><span>${beer.description}</span>`);
  const firstBrewed = createElement('div', { class: 'firstBrewed' }, `<em>first brewed:</em><span>${beer.first_brewed}</span>`);
  return [tagLine, firstBrewed, description];
};

export const beerItem = (beer) => {
  const beerDetails = createElement('div', { class: 'beerDetails' });
  const beerDetailsHeader = createElement('h2', { class: 'beerDetailsHeader' }, beer.name);
  const beerDetailsBody = createElement('div', { class: 'beerDetailsBody' });
  addElementsToElement(beerDetailsBody, beerDetailsBodyContent(beer));
  addElementsToElement(beerDetails, [beerDetailsHeader, beerDetailsBody]);
  beerDetailsHeader.onclick = () => onClickAccordion(beer.id);
  return [beerDetails];
};

export const beersList = beers => beers.map((beer) => {
  const listItem = createElement('li', { id: `accordion${beer.id}`, class: 'accordion' });
  addElementsToElement(listItem, beerItem(beer));
  return listItem;
});

export const beersAccordion = (beers) => {
  const header = createElement('h2', {}, 'Beers');
  const listContainer = createElement('ul', {});
  addElementsToElement(listContainer, beersList(beers));
  return [header, listContainer];
};

export const createBeersAccordion = beers => addElementsToAccordion(beersAccordion(beers));