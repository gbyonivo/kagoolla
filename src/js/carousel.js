
import jquery from 'jquery';
import 'slick-carousel';
import {
  createElement,
  addElementsToElement,
  getElementById
} from '../functions/document';
import { store } from '../data';


export const beerImg = (beer) => {
  const beerImageBox = createElement('div', { class: 'carouselImageBox' });
  const beerImage = createElement('img', { class: 'carouselImage', src: beer.image_url });
  addElementsToElement(beerImageBox, [beerImage]);
  return beerImageBox;
};

export const beerDetails = beer => `
  <h2>${beer.name}</h2>
  <div>${beer.first_brewed}</div>
  <div>${beer.tagline}</div>
  <div>${beer.description}</div>
  <div>Alcohol Volume: ${beer.abv}</div>
`;

export const beerItem = (beer) => {
  const beerBox = createElement('div', { class: 'carouselItemContent' });
  const beerName = createElement('div', { class: 'carouselDetails' }, beerDetails(beer));
  const beerImageBox = beerImg(beer);
  addElementsToElement(beerBox, [beerImageBox, beerName]);
  return [beerBox];
};

export const beersList = beers => beers.map((beer) => {
  const listItem = createElement('div', { id: `carousel${beer.id}`, class: 'carouselItem' });
  addElementsToElement(listItem, beerItem(beer));
  return listItem;
});

export const createBeersCarousel = (settings, id) => {
  addElementsToElement(getElementById(id), beersList(store.getCarouselBeers()));
  jquery(`#${id}`).slick(settings);
};