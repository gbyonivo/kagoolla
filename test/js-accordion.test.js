import { mount } from 'enzyme';
import * as accordion from '../src/js/accordion';

const beers = [{
  name: 'Egyptian King',
  id: 1,
  description: 'goals galore',
  first_brewed: 'heaven',
  tagline: 'pyramids'
}];

describe('accordion.createBeersAccordion', () => {
  it('should create a accordion', () => {
    document.body.innerHTML = '<div id="accordionWrapper"></div>';
    accordion.createBeersAccordion([]);
    expect(document.body).toMatchSnapshot();
  });
});

describe('accordion.beersAccordion', () => {
  it('should create a beersAccordion', () => {
    const [header, listContainer] = accordion.beersAccordion([]);
    expect(header).toMatchSnapshot();
    expect(listContainer).toMatchSnapshot();
  });
});

describe('accordion.beersList', () => {
  it('should create a beersList', () => {
    const beersList = accordion.beersList(beers);
    expect(beersList[0]).toMatchSnapshot();
  });
});

describe('accordion.beerItem', () => {
  it('should create a beerItem', () => {
    const [beerItem] = accordion.beerItem(beers[0]);
    expect(beerItem).toMatchSnapshot();
  });
});
