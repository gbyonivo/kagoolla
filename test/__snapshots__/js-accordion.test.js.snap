// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accordion.beerItem should create a beerItem 1`] = `
<div
  class="beerDetails"
>
  <h2
    class="beerDetailsHeader"
  >
    Egyptian King
  </h2>
  <div
    class="beerDetailsBody"
  >
    <div
      class="tagLine"
    >
      <em>
        tag line:
      </em>
      <span>
        pyramids
      </span>
    </div>
    <div
      class="firstBrewed"
    >
      <em>
        first brewed:
      </em>
      <span>
        heaven
      </span>
    </div>
    <div
      class="description"
    >
      <em>
        description:
      </em>
      <span>
        goals galore
      </span>
    </div>
  </div>
</div>
`;

exports[`accordion.beersAccordion should create a beersAccordion 1`] = `
<h2
  class="accordionHeader"
>
  Available Beers
</h2>
`;

exports[`accordion.beersAccordion should create a beersAccordion 2`] = `<ul />`;

exports[`accordion.beersList should create a beersList 1`] = `
<li
  class="accordion"
  id="accordion1"
>
  <div
    class="beerDetails"
  >
    <h2
      class="beerDetailsHeader"
    >
      Egyptian King
    </h2>
    <div
      class="beerDetailsBody"
    >
      <div
        class="tagLine"
      >
        <em>
          tag line:
        </em>
        <span>
          pyramids
        </span>
      </div>
      <div
        class="firstBrewed"
      >
        <em>
          first brewed:
        </em>
        <span>
          heaven
        </span>
      </div>
      <div
        class="description"
      >
        <em>
          description:
        </em>
        <span>
          goals galore
        </span>
      </div>
    </div>
  </div>
</li>
`;

exports[`accordion.createBeersAccordion should create a accordion 1`] = `
<body>
  <div
    id="accordionWrapper"
  >
    <h2
      class="accordionHeader"
    >
      Available Beers
    </h2>
    <ul />
  </div>
</body>
`;
